src/item_tname.cpp

```diff
@@ -119,6 +119,44 @@ std::string burn( item const &it, unsigned int /* quantity */,
     return {};
 }
 
+std::string custom_item_prefix( item const &it, unsigned int /* quantity */,
+                                segment_bitset const &/* segments */ )
+{
+    std::string prefix;
+    // item flags
+    for( const flag_id &f : it.get_flags() ) {
+        if( !f->item_prefix().empty() ) {
+            prefix += f->item_prefix();
+        }
+    }
+    // item type flags
+    for( const flag_id &f : it.type->get_flags() ) {
+        if( !f->item_prefix().empty() ) {
+            prefix += f->item_prefix();
+        }
+    }
```

Comment:
Is it possible for `it.get_flags()` and `it.type->get_flags()` to contain identical flags? If so the flags should be merged into a single set before iteration.